### ISSUE TITLE
Updating from 1.0.2 to 1.1.4 from the Add-ons panel causes mmd_tools to be unavailable with errors

### ISSUE THREAD

**Issue by listelin** [UuuNyaa/blender_mmd_tools#14]
### 1.0.2 to 1.1.4
1.1.4
![114](https://user-images.githubusercontent.com/44924233/147640249-9113fc2a-e799-4087-abcd-e88fc4fb22ce.jpg)

no panel 
![panel](https://user-images.githubusercontent.com/44924233/147640261-f7b48ce9-67b8-4df0-b240-a765b459d0f5.jpg)

```
module changed on disk: 'C:\\Users\\a\\AppData\\Roaming\\Blender Foundation\\Blender\\3.0\\scripts\\addons\\mmd_tools\\__init__.py' reloading...
TypeError: PointerProperty(...) expected an RNA type, failed with: RuntimeError: , missing bl_rna attribute from 'RNAMetaPropGroup' instance (may not be registered)

Exception in module register(): C:\Users\a\AppData\Roaming\Blender Foundation\Blender\3.0\scripts\addons\mmd_tools\__init__.py
Traceback (most recent call last):
  File "C:\blender\blender-3.0.0-windows-x64\3.0\scripts\modules\addon_utils.py", line 387, in enable
    mod.register()
  File "C:\Users\a\AppData\Roaming\Blender Foundation\Blender\3.0\scripts\addons\mmd_tools\__init__.py", line 228, in register
    bpy.utils.register_class(cls)
ValueError: bpy_struct "MMDRoot" registration error: 'translation' PointerProperty could not register (see previous error)
```

# Reproduction Procedure
1. Delete mmd_tools addon
2. Download released version mmd_tools (1.0.2) and install , restart
3. Update main branch from preference/ mmd_tools addon

**Comment by UuuNyaa** [comment]
Thank you for the report.

This is a known issue due to a change in the structure of the MMDRoot property.
We have two choices.
1. overwrite the Startup File with the Factory Settings
    ![image](https://user-images.githubusercontent.com/70152495/147649931-df074aad-97c2-4dd6-b7ee-d21a6fdc564f.png)
2. give up on the MMD property hierarchy rules
    Make the translation property, which should be placed in the MMD root empty object, independent as a separate property.
    https://github.com/UuuNyaa/blender_mmd_tools/commit/de2ccf2c824a8593b113d163db5befaa3c1a448d#diff-f84e0fa5179c0d2ca111c022e237362c3fa3cd785b71f802b8f7d6e8acd67edbL57

**Comment by listelin** [comment]
I tried the following two things about step 1, but the error was not resolved.

### clear and 1.0.2 and update
1. overwrite the Startup File with the Factory Settings, quit blender
2. remove mmd_tools from addon directory
3. install 1.0.2 and update -> same problem occurs

### clear and install 1.1.4 zip
1. overwrite the Startup File with the Factory Settings, quit blender
2. remove mmd_tools from addon directory
3. make zip from working mmd_tools 1.1.4 code , and install that zip -> same problem occurs

I'm sorry I don't understand what the problem is, but I'll report it.

**Comment by UuuNyaa** [comment]
Hmmm.... 🤔
Since there are two types of translation property structures, the problem occurs when upgrading across them.
Blender seems to record the structure of the translation properties somewhere other than in the Startup File.

We can't force users to do "5 completery delete 3.0 setting folder", so I'll make the translation property independent.

**Comment by listelin** [comment]
I'm still trying to figure out what is the problem
 But in another environment where I did update 1.0.2 ->... →1.1.3→1.1.4 was fine,
 so I copied the userpref.blend and startup.blend from that environment to and it works fine.

**Comment by UuuNyaa** [comment]
There may be a problem with the reset steps.

Please try the following steps:
1. Remove old mmd_tools from Blender
2. Restart Blender
3. Restore the Startup File to Factory Settings
4. Install new mmd_tools

**Comment by listelin** [comment]
> * Remove old mmd_tools from Blender
> * Restart Blender
> * Restore the Startup File to Factory Settings
> * Install new mmd_tools

That works in 1.0.2.
But updating to main branch (1.1.4) cause same problem.

The above NG also works with 1.0.2, although I didn't explain it well enough, the problem was that Update was impossible.

**Comment by UuuNyaa** [comment]
I am not able to reproduce the problem in my environment.
Can you reproduce the problem by **just** installing and updating the add-on?

**Comment by listelin** [comment]
> I am not able to reproduce the problem in my environment. Can you reproduce the problem by **just** installing and updating the add-on?

Yes
1. remove mmd_tools
2. restart blender
3. install 1.0.2 and enable -> OK with 1.0.2
4. check mmd_tools add on update  from preference, target: main , press Upate , restart blender
5. error occurs and 1.1.4 doesn't work as I wrote in the beginning.

It doesn't reproduce, does it?
Then my startup.blend or userpref.blend may be the cause of the problem....

**Comment by nagadomi** [comment]
This may be a problem with the registration order of the classes.
`MMDTranslation` need to be `register()` before `MMDRoot`.

It happened once when I installed mmd_tools from git clone, but once I restarted blender, it didn't happen for unknown reason.

**Comment by UuuNyaa** [comment]
@listelin @nagadomi Thanks for information.

> This may be a problem with the registration order of the classes.

There is that possibility.

MMD **UuuNyaa** Tools has a mechanism to sort before classes register.
However, MMD Tools does not have the mechanism.

https://github.com/UuuNyaa/blender_mmd_uuunyaa_tools/blob/64597057ee947f685d80f14a1bb9e5517dde41b2/mmd_uuunyaa_tools/auto_load.py#L72

I'll look into it more.

**Comment by UuuNyaa** [comment]
I added code in v1.1.5 to sort the class dependencies.

**Comment by listelin** [comment]
It may still be in progress, but this is the result of updating to 1.1.5.
I installed 1.0.2 and made it available, then updated with check mmd_tools addon update.

![mmd](https://user-images.githubusercontent.com/44924233/147672674-18dea135-3e0e-42ca-b3b0-7c58e7d0c643.jpg)

![UI](https://user-images.githubusercontent.com/44924233/147673097-1cba9867-4206-4f56-87c9-843bdf744578.jpg)
Scene setup and Material sorter appears. other have no content panel,
and I get a similar error and cannot use mmd_tools.
```
module changed on disk: 'C:\\Users\\a\\AppData\\Roaming\\Blender Foundation\\Blender\\3.0\\scripts\\addons\\mmd_tools\\__init__.py' reloading...
TypeError: PointerProperty(...) expected an RNA type, failed with: RuntimeError: , missing bl_rna attribute from 'RNAMetaPropGroup' instance (may not be registered)

Exception in module register(): C:\Users\a\AppData\Roaming\Blender Foundation\Blender\3.0\scripts\addons\mmd_tools\__init__.py
Traceback (most recent call last):
  File "C:\blender\blender-3.0.0-windows-x64\3.0\scripts\modules\addon_utils.py", line 387, in enable
    mod.register()
  File "C:\Users\a\AppData\Roaming\Blender Foundation\Blender\3.0\scripts\addons\mmd_tools\__init__.py", line 239, in register
    bpy.utils.register_class(cls)
ValueError: bpy_struct "MMDRoot" registration error: 'translation' PointerProperty could not register (see previous error)
Python: Traceback (most recent call last):
  File "C:\Users\a\AppData\Roaming\Blender Foundation\Blender\3.0\scripts\addons\mmd_tools\panels\tool.py", line 57, in draw
    root = mmd_model.Model.findRoot(active_obj)
  File "C:\Users\a\AppData\Roaming\Blender Foundation\Blender\3.0\scripts\addons\mmd_tools\core\model.py", line 161, in findRoot
    return FnModel.find_root(obj)
  File "C:\Users\a\AppData\Roaming\Blender Foundation\Blender\3.0\scripts\addons\mmd_tools\core\model.py", line 24, in find_root
    if obj.mmd_type == 'ROOT':
AttributeError: 'Object' object has no attribute 'mmd_type'

location: <unknown location>:-1
```

**Comment by nagadomi** [comment]
Does it happen if you restart blender after updating the addon?

1. install 1.0.2
2. update to the latest version (**ignore any errors in this step**)
3. restart blender
3. check that the addon works fine (enable the addon if it is not enabled)

No errors on my pc.

**Comment by nagadomi** [comment]
@listelin 
Do you have cats-blender-plugin installed? It conflicts with mmd_tools because it has the old mmd_tools source code locally.

**Comment by listelin** [comment]
> * install 1.0.2
> * update to the latest version (**ignore any errors in this step**)
At this point, preference broken, and no update button

> * restart blender
> * check that the addon works fine (enable the addon if it is not enabled)

So, once restart and update. same as above erros.

> @listelin Do you have cats-blender-plugin installed? It conflicts with mmd_tools because it has the old mmd_tools source code locally.

I've heard of it, but have never installed it.

I have two Windows PCs, and I can reproduce the same problem over and over again.
If it doesn't happen in other environments. The add-on may be suspect ?

This is all addon enabled
![addons](https://user-images.githubusercontent.com/44924233/147686767-d40661aa-f48b-46e0-ae4a-579763b032ae.jpg)

and disable these and try again
- pose library
- icon viewer
- corrective shapekey
- material utility
- paint palette

but same problem occurs.

PC1: blender 3.0 1.1.3 -> 1.0.2 -> 1.1.4 -> 1.1.5 : never updatable
PC2: blender 3.0 1.1.3 -> 1.0.2 -> 1.1.4 -> 1.1.5 : never updatable
PC1: blender 3.1 1.1.3 -> 1.1.4 -> 1.1.5 : no problem ( same addons)

I wonder what's going on.

**Comment by listelin** [comment]
If other people can't update from 1.0.2, I think it needs to be addressed.
If the problem is only in my environment, I can use blender 3.1 alpha, so I think you can lower the priority of this problem.

**Comment by UuuNyaa** [comment]
@listelin Thanks for doing so much testing.
I have only tested it on Linux, so it may be a problem only in Windows.
Or maybe there is a problem specific to Blender 3.0.

> If the problem is only in my environment

I'm sure it will reproduce in someone else's environment.

@nagadomi Is the OS of your test environment Windows?

**Comment by UuuNyaa** [comment]
I tested the version transition in the following order, and was able to upgrade without any problems.
`none` -> `v1.1.3` -> `v1.0.2` -> `v1.1.6`

https://user-images.githubusercontent.com/70152495/147695224-91ba179b-3e0d-432c-bd2d-adb598c6b4c9.mp4

Is there anything different from your steps?

**Comment by nagadomi** [comment]
@UuuNyaa 
I tested it on both Linux+Blender 2.93 and Windows 10+Blender 3.0 (zip version, 1.02 -> 1.1.6(main branch) only).
No errors.

**Comment by UuuNyaa** [comment]
@nagadomi Thank you for your information.
The mystery deepens... 😖

**Comment by listelin** [comment]
Solved in 1.1.6!!
Thank you! @UuuNyaa @nagadomi

**Comment by UuuNyaa** [comment]
@listelin Yay 🥳
Thank you for your cooperation.
